#! /bin/sh /usr/share/dpatch/dpatch-run

## 06_call_udevsettle
## by Reinhard Tartler <siretart@ubuntu.com
##
## DP: See https://bugs.launchpad.net/bugs/132373

--- cryptsetup-1.0.5/lib/libdevmapper.c	2007-05-29 11:45:03 +0000
+++ cryptsetup-1.0.5/lib/libdevmapper.c	2007-11-04 18:40:37 +0000
@@ -184,6 +184,21 @@
 	if (dmi.read_only)
 		options->flags |= CRYPT_FLAG_READONLY;
 
+	/*
+	 * patch inspired from
+	 * https://bugzilla.novell.com/show_bug.cgi?id=285478
+	 *
+	 * At this point, it is quite likely that there are currently
+	 * devmapper events being processed. In order to avoid
+	 * workarounds (which involve calling udevsettle at the
+	 * 'right' places in scripts integrating cryptsetup), we are
+	 * checking here if udevsettle is available and call it if it
+	 * is.
+	 */
+	if (0 == access("/sbin/udevadm", X_OK)) {
+		system("/sbin/udevadm settle");
+	}
+	
 	r = 0;
 	
 out:

